### DISCUSSION TITLE
Magnetic field model updates

### DISCUSSION THREAD

**Discussion by jvrod23** [spacepy/spacepy#576] [Support and Q&A]
I'm using irbempy in a study of Metop data from November 2021.  Related to internal and external magnetic field models and their drivers, the following questions have come up:

1. Which update of the IGRF is SpacePy currently using?  
2. I updated the QinDenton file (omnidata.h5) using toolbox.update.  The last data in the file is from 2021-07-14.  Is an update of the QD file expected?  (I was able to work around this by creating an 'omnivals' file with the correct keys - I only need Kp for now.  I couldn't find any documentation on the format/contents of such a file but found the keys in the code and went from there.)
3. I tried to use the updated omni2data.h5 as an alternative (last date is 2022-02-04) but stopped after realizing that the keys are different from the QD file.  I did notice that the 'epoch' validmax and scalemax in the omni2data.h5 file are 2020-12-31T23:59:59, earlier than the last date in the file, which could prevent post-2020 data from being read since the epoch would be invalid.

Thanks.

**Comment by drsteve** [comment]
1. SpacePy is using IGRF 13 (accessed through IRBEM, where I committed the update to IGRF 13 in April 2020). Not that you're probably using it, but (for the sake of any wandering readers) the coordinate transform updates to use a Python backend in #296 also use IGRF13.
2. The updates are pulled from the Van Allen Probes ECT science data center, which I guess is not running the Qin-Denton processing right now? I believe I have login credentials for the right machine, so I can push some updated data as a short term fix. As a longer term fix I'll try to find some time to figure out why the process isn't running over there. In the much longer term that processing is going to get moved and should be maintained by folks who are paid to maintain it...
3. This sounds like we need to fix the metadata for the OMNI2 epoch. I'm not too sure whether this is an upstream issue or whether it's our fault, but one of us should file this as a SpacePy issue. If it's not our fault then we can tag it as upstream and include a workaround. If it is our fault we can just fix.

**Reply by drsteve** [reply]
To update on point 2: I apparently don't have access at the moment, but we've pushed data through January 19th 2022 so please check to see if that helps. Getting the processing running again could be anywhere from a 10 minute job to a few hour job, so it's on the _to do_ list but I can't promise any particular date it'll be back up by.
